crypto/asn1/asn1_parse.c

```diff
@@ -115,9 +117,7 @@ static int asn1_parse2(BIO *bp, const unsigned char **pp, long length,
         op = p;
         j = ASN1_get_object(&p, &len, &tag, &xclass, length);
         if (j & 0x80) {
-            if (BIO_write(bp, "Error in encoding\n", 18) <= 0)
-                goto end;
-            ret = 0;
+            BIO_puts(bp, "Error in encoding\n");
```

Comment:
what happened to the if here?

Reply:
It felt needless and was convinced to disappear :)

Reply:
It is inconsistent with the rest of the function..
(Although it also doesnt check a printf either)

Reply:
I think it is more consistent now than it was. The principle to follow here is:
for printing error messages, the return value should be ignored, but for regular printing it should not.